tests/test_tra.py

```diff
@@ -223,6 +223,57 @@ def _get_args(self):
     def _get_threshold(self):
         return 0.6
 
+    @testing_utils.retry(ntries=3)
+    def test_eval_fixed_label_not_in_cands(self):
+        # test where cands during eval do not contain test label
+        args = self._get_args()
+        args[
+            'model'
+        ] = 'parlai.agents.transformer.polyencoder:IRFriendlyPolyencoderAgent'
+        args['eval_candidates'] = 'fixed'
+
+        teacher = CandidateTeacher({'datatype': 'train'})
+        all_cands = teacher.train + teacher.val + teacher.test
+        train_val_cands = teacher.train + teacher.val
+        all_cands_str = '\n'.join([' '.join(x) for x in all_cands])
+        train_val_cands_str = '\n'.join([' '.join(x) for x in train_val_cands])
+
+        with testing_utils.tempdir() as tmpdir:
+            tmp_cands_file = os.path.join(tmpdir, 'all_cands.text')
+            with open(tmp_cands_file, 'w') as f:
+                f.write(all_cands_str)
+            tmp_train_val_cands_file = os.path.join(tmpdir, 'train_val_cands.text')
+            with open(tmp_train_val_cands_file, 'w') as f:
+                f.write(train_val_cands_str)
+            args['fixed_candidates_path'] = tmp_cands_file
+            args['encode_candidate_vecs'] = False  # don't encode before training
+            args['ignore_bad_candidates'] = False
+            args['model_file'] = os.path.join(tmpdir, 'model')
+            args['dict_file'] = os.path.join(tmpdir, 'model.dict')
+            args['num_epochs'] = 20
+            # Train model where it has access to the candidate in labels
+            stdout, valid, test = testing_utils.train_model(args)
+            self.assertGreaterEqual(
```

Comment:
Is our goal to test that it performs at a certain level with the fixed candidates, or that the fixed candidate can be loaded/encoded correctly, or what? Because if you dropped this greaterEqual, then you could drop the retry, and I think this test would still be adequate for what we want, no?

Reply:
Yeah we just want to check that the fixed cands can be loaded/encoded correctly. I will keep greaterEqual test but i will drop it to 0.0 and get rid of retry

Reply:
You can just drop the adsert then. It will still run